finagle-core/src/main/scala/com/twitter/finagle/tracing/Id.scala

```diff
@@ -114,11 +131,36 @@ object TraceId {
         if (parent64 == span64) None else Some(SpanId(parent64)),
         SpanId(span64),
         sampled,
-        flags
+        flags,
+        Some(SpanId(traceIdHigh))
       )
       Return(traceId)
     }
   }
+
+  /**
+   * Extracts the high 64bits (if set) and low 64bits from a B3 TraceID's string representation.
+   *
+   * @param spanId A 64bit or 128bit Trace ID.
+   *
+   * @return An <code>(Option[SpanId], Option[SpanId])</code> of 64bit or 128bit TraceID. Structure is (high, low).
+   */
+  def mk128BitTraceId(spanId: String): (Option[SpanId], Option[SpanId]) = {
```

Comment:
Thanks. I meant to mark that as another question (conventions for creating one-off case classes like this) but forgot to mention it in the PR.

Reply:
Was just getting around to this, and remembered a major part of the question. I wanted to call that case class just "TraceId", but this is already taken from the case class encapsulating the traceId, parentId, spanId, etc. In other implementations, I think I've seen the current `TraceId` named things like TraceContext or SpanContext. I had considered a case class named`FullTraceId`, but I was concerned this may cause confusion with the existing `TraceId` type.

Since a simple "TraceId" is already taken, are there any suggestions for naming here?

Reply:
if this were a java builder, I would just say something like `TraceId.Builder.parseTraceId(String)` or similar, which internally sets one of the two. Maybe you could consider making a constructor of TraceId which takes a string form of TraceId and internally does the same?

Reply:
I agree that the naming is now confusing at best, but don't think its something that should be addressed in this PR. If we want to take the easy route out of this, how about `SpanId128`. Might want to rename the method as well, since it seems (at least to me, happy to hear why I'm wrong) that we're really dealing in span-ids.